apps/deploy-web/src/components/settings/AutoTopUpSetting/AutoTopUpSetting.tsx

```diff
@@ -0,0 +1,216 @@
+
+  const defaultLimitValues = useMemo(() => {
+    return fields.reduce(
+      (acc, field) => {
+        acc[field] = uaktToAKT(props[field] || 0);
+        return acc;
+      },
+      {} as Record<LimitFields, number>
+    );
+  }, [props]);
+
+  const form = useForm<z.infer<typeof formSchema>>({
+    defaultValues: {
+      ...defaultLimitValues,
+      expiration: format(expiration || addYears(new Date(), 1), "yyyy-MM-dd'T'HH:mm")
+    },
+    resolver: zodResolver(formSchema)
+  });
+  const { handleSubmit, control, setValue, reset } = form;
+
+  useEffect(() => {
+    setValue("uaktFeeLimit", uaktToAKT(props.uaktFeeLimit || 0));
+  }, [props.uaktFeeLimit]);
+
+  useEffect(() => {
+    setValue("usdcFeeLimit", uaktToAKT(props.usdcFeeLimit || 0));
+  }, [props.usdcFeeLimit]);
+
+  useEffect(() => {
+    setValue("uaktDeploymentLimit", uaktToAKT(props.uaktDeploymentLimit || 0));
+  }, [props.uaktDeploymentLimit]);
+
+  useEffect(() => {
+    setValue("usdcDeploymentLimit", uaktToAKT(props.usdcDeploymentLimit || 0));
+  }, [props.usdcDeploymentLimit]);
+
+  useEffect(() => {
+    if (expiration) {
+      setValue("expiration", format(expiration || addYears(new Date(), 1), "yyyy-MM-dd'T'HH:mm"));
+    }
+  }, [expiration]);
+
+  const execSubmitterRoleAction: SubmitHandler<FormValues> = useCallback(
+    async (next: FormValues, event: React.BaseSyntheticEvent<SubmitEvent>) => {
+      const role = event.nativeEvent.submitter?.getAttribute("data-role");
+      await onSubmit(role as "revoke-all" | "update", convertToUakt(next));
+      reset(next);
+    },
+    [onSubmit, reset]
+  );
+
+  return (
+    <div>
+      <Form {...form}>
+        <form onSubmit={handleSubmit(execSubmitterRoleAction)} noValidate>
+          <div className="flex">
+            <div className="flex-1">
+              <FormField
+                control={control}
+                name="uaktDeploymentLimit"
+                render={({ field, fieldState }) => {
+                  return <FormInput {...field} dirty={fieldState.isDirty} type="number" label="AKT Deployments Limit, AKT" min={0} step={0.000001} />;
+                }}
+              />
+            </div>
+
+            <div className="ml-3 flex-1">
+              <FormField
+                control={control}
+                name="uaktFeeLimit"
+                render={({ field, fieldState }) => {
+                  return <FormInput {...field} dirty={fieldState.isDirty} type="number" label="AKT Fees Limit, AKT" min={0} step={0.000001} />;
+                }}
+              />
+            </div>
+          </div>
+
+          <div className="flex">
+            <div className="flex-1">
+              <FormField
+                control={control}
+                name="usdcDeploymentLimit"
+                render={({ field, fieldState }) => {
+                  return <FormInput {...field} dirty={fieldState.isDirty} type="number" label="USDC Deployments Limit, AKT" min={0} step={0.000001} />;
```

Comment:
Should the label say `AKT` for usdc as well?

Reply:
it does right there `USDC Deployments Limit, ---> AKT <---`
or wdum?